utils/managed_bytes.hh

```diff
@@ -157,21 +265,27 @@ public:
     managed_bytes(std::initializer_list<bytes::value_type> b) : managed_bytes(b.begin(), b.size()) {}
 
     ~managed_bytes() noexcept {
-        if (external()) {
-            free_chain(_u.ptr);
+        if (is_multi_chunk()) {
+            free_chain(_u.multi_chunk_ref);
+        } else if (is_single_chunk()) {
+            auto& alctr = current_allocator();
+            alctr.destroy(_u.single_chunk_ref.ptr);
         }
     }
 
+    // Defined later in the file because it depends on managed_bytes_mutable_view.
     managed_bytes(const managed_bytes& o);
 
-    managed_bytes(managed_bytes&& o) noexcept
-        : _u(o._u)
-    {
-        if (external()) {
-            // _u.ptr cannot be null
-            _u.ptr->backref = &_u.ptr;
+    managed_bytes(managed_bytes&& o) noexcept {
+        // Microoptimization: we use memcpy instead of assigments because
+        // the compiler refuses the merge the load/stores otherwise for some reason.
+        std::memcpy(reinterpret_cast<char*>(this), &o, sizeof(managed_bytes));
```

Comment:
Don't know if it's completely legal. I'll look the other way.

Reply:
Only trivial types are involved, so copying object representation should be equivalent to default copy constructors.
But I'm not sure if it's completely legal either, union rules are too hard to interpret.

If we are concerned about this, we can just use `char[15]` instead of a union, and access the "union fields" manually via memcpy.